models/tenant.py

```diff
@@ -16,17 +17,24 @@ class TenantModel(BaseModel):
     # relationships
     staff = relationship("UserModel", secondary="staff_tenant_links")
     leases = db.relationship(
-        "LeaseModel", backref="tenant", lazy=True, cascade="all, delete-orphan"
+        "LeaseModel",
+        backref="tenant",
+        # lazy="dynamic" is required so that we can filter() over leases in
+        # json(). See: https://stackoverflow.com/questions/11578070
+        lazy="dynamic",
+        cascade="all, delete-orphan",
     )
 
     def json(self):
+        first_active_lease = self.leases.filter(LeaseModel.active()).first()
+        active_lease_json = first_active_lease.json() if first_active_lease else ""
```

Comment:
It seems that we want to eagerly load this instead of lazily. Perhaps it would be more correct to eagerly load a custom filtered collection result.
https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#using-contains-eager-to-load-a-custom-filtered-collection-result

I'm not sure if this would work, but it sounds like this might be what we want. I'm not opposed to the solution you provided either, just providing an alternate method that has the possibility to only hit the database once, but this might be more work and might not work with the `User.find` method.

Reply:
I'm curious if removing the lazy argument works. Then we should be able to use filter on the returned data. Since it might default to eagerly loading. I'm not sure though, I'm not that well read up on sqlalchemy. I just don't think we should lazily load the data when we know we will be using it all the time.

Reply:
Removing lazy argument does not work. Errors look like this:
```
FAILED tests/integration/test_properties.py::test_get_property_by_id - AttributeError: 'InstrumentedList' object has no attribute 'filter'
```
I'm honestly still a little confused by the docs; seems like there are other options other than `dynamic`, and I'm unsure if `dynamic` is the only option that will work here. Maybe there's something better, but maybe those things would require us to restructure other things. For now, this works---if there's something especially egregious about this, someone should call it out, but otherwise we'll let this lie!